presto-product-tests/src/main/java/io/prestosql/tests/hive/TestHiveViews.java

```diff
@@ -295,6 +297,40 @@ public void testSelectFromHiveViewWithoutDefaultCatalogAndSchema()
                 .containsExactly(row(1L));
     }
 
+    @Test
+    public void testTimestampHiveView() {
+        onHive().executeQuery("DROP TABLE IF EXISTS timestamp_hive_table");
+        onHive().executeQuery("CREATE TABLE timestamp_hive_table AS SELECT cast('1990-01-02 12:13:14.123456789' AS timestamp) ts");
+        onHive().executeQuery("DROP VIEW IF EXISTS timestamp_hive_view");
+        onHive().executeQuery("CREATE VIEW timestamp_hive_view AS SELECT * FROM timestamp_hive_table");
+
+        // timestamp_precision not set UNSET
+        assertThat(query("SELECT CAST(ts as VARCHAR) FROM timestamp_hive_view")).containsExactly(row("1990-01-02 12:13:14.123"));
+        assertThatThrownBy(
+                // TODO it is not possible to query Hive view with timestamps if hive.timestamp-precision=NANOSECONDS
+                () -> assertThat(query("SELECT CAST(ts as VARCHAR) FROM hive_timestamp_nanos.default.timestamp_hive_view")).containsExactly(row("1990-01-02 12:13:14.123456789"))
+        ).hasMessageContaining("timestamp(9) projected from query view at position 0 cannot be coerced to column [ts] of type timestamp(3) stored in view definition");
+
+        setSessionProperty("hive.timestamp_precision", "'MILLISECONDS'");
+        setSessionProperty("hive_timestamp_nanos.timestamp_precision", "'MILLISECONDS'");
+        assertThat(query("SELECT CAST(ts as VARCHAR) FROM timestamp_hive_view")).containsExactly(row("1990-01-02 12:13:14.123"));
+        assertThatThrownBy(
+                // TODO it is not possible to query Hive view with timestamps if hive.timestamp-precision=NANOSECONDS
+                () -> assertThat(query("SELECT CAST(ts as VARCHAR) FROM hive_timestamp_nanos.default.timestamp_hive_view")).containsExactly(row("1990-01-02 12:13:14.123"))
+        ).hasMessageContaining("timestamp(9) projected from query view at position 0 cannot be coerced to column [ts] of type timestamp(3) stored in view definition");
+
+        setSessionProperty("hive.timestamp_precision", "'NANOSECONDS'");
+        setSessionProperty("hive_timestamp_nanos.timestamp_precision", "'NANOSECONDS'");
+        assertThatThrownBy(
+                // TODO timestamp_precision has no effect on Hive views
+                () -> assertThat(query("SELECT CAST(ts as VARCHAR) FROM timestamp_hive_view")).containsExactly(row("1990-01-02 12:13:14.123456789"))
+        ).hasMessageContaining("0 - actual:   1990-01-02 12:13:14.123|");
```

Comment:
include both "actual" and "expect" in this message.
or, change to plan not-nested asssertion and use the actual wrong value in assertion, and put the expected value in a TODO comment

Reply:
Actually I like this one more. I would rather assert for what it feels should happen and only wrap it in temporary `assertThatThrownBy`.
As for putting `1990-01-02 12:13:14.123456789` in expected message it does not give us much. There is only one row expected anyway. And we will just make test definition longer by adding that. Do you insist?

Reply:
NVM. Changed.